libraries/browser-tracker-core/src/tracker/out_queue.ts

```diff
@@ -422,6 +435,22 @@ export function OutQueueManager(
     }
   }
 
+  function shouldRetryForStatusCode(statusCode: number) {
+    // success, don't retry
+    if (statusCode >= 200 && statusCode < 300) {
+      return false;
+    }
+
+    // retry if status code among custom user-supplied retry codes
+    if (retryStatusCodes.includes(statusCode)) {
+      return true;
+    }
+
+    // retry if status code *not* among custom user-supplied don't retry codes and default don't retry codes
+    const dontRetryStatusCodesDefault = [400, 401, 403, 410, 422];
```

Comment:
We're redefining this constant on every call to this function. Could we hoist it out of the function? 

Maybe even do something fancy near the config params defaults bit in `libraries/browser-tracker-core/src/tracker/index.ts`?